### PR TITLE
93 make mentor page responsive based on backend data

### PR DESCRIPTION
**Pull request by nathnael-desta**
Make the mentor page responsive to the data that is fetched for the backend. 
Since the backend hasn't finished the conversations db, i made a json-server db for the fetching.

### CODE REVIEW

**Comment by nathnael-desta** [comment]
Screenshot from 2024-12-18 10-15-51
on wide screen

**Comment by nathnael-desta** [comment]
Screenshot from 2024-12-18 10-16-11
on smaller screen

**Comment by nathnael-desta** [comment]
Screenshot from 2024-12-18 10-16-39
profile shadcn sheet

File: components/shared/Mentor/Chat.tsx [hunk 1]
```diff
@@ -13,59 +13,85 @@ import { ChatProps, threadType } from "@/types/mentor";
...
+      // Find the current id from props
+      if (!userMessages) {
+        return;
+      }
+
+      // Append the new message to the messages array
+      const updatedMessages = [...userMessages.messages, newMessage];
+
+      // Update the backend
+      const patchResponse = await fetch(`http://localhost:3001/userMessages/${userMessages.id}`, {
```
Comment (misudesu) [inline components/shared/Mentor/Chat.tsx:45]: 💡use .env file.  
💡use general api request end point in front end side. i mean reuse it.

File: components/shared/Mentor/Chat.tsx [hunk 2]
```diff
@@ -13,59 +13,85 @@ import { ChatProps, threadType } from "@/types/mentor";
...
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ messages: updatedMessages }),
+      });
+
+      console.log("the patch response", patchResponse)
```
Comment (misudesu) [inline components/shared/Mentor/Chat.tsx:53]: 😒 how you handle error ?

File: components/shared/Mentor/mentor-container.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ const MentorContainer = ({ users }: MentorContainerProps) => {
       try {
         // fetches the userDetails from db
         const response = await fetch(
-          `http://localhost:3001/userDetails?userId=${currentUser.userId}`
+          `http://localhost:3001/userDetails?id=${currentUser.id}`
```
Comment (misudesu) [inline components/shared/Mentor/mentor-container.tsx:21]: here also .env and base api request

File: components/shared/Mentor/mentor-container.tsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ const MentorContainer = ({ users }: MentorContainerProps) => {
...
         );
 
         if (!response.ok) {
```
Comment (misudesu) [inline components/shared/Mentor/mentor-container.tsx:24]: does it make sense "Network response was not ok"
do you have a plan ? to  handle this error message throw new Error("Network response was not ok");

**Review by misudesu — CHANGES_REQUESTED** [review]

**Comment by misudesu** [comment]
check conflicts

File: components/shared/Mentor/Chat.tsx (hunk 1 above)
Comment (nathnael-desta) [reply components/shared/Mentor/Chat.tsx]: i have fixed it in a new commit, i put the endpoints into a file and linked them like you showed me

File: components/shared/Mentor/Chat.tsx (hunk 2 above)
Comment (nathnael-desta) [reply components/shared/Mentor/Chat.tsx:54]: I was using a json server to mock the backend response but now I have been informed that the backend schma will change a lot from my implementation here and i will need to completely redo the fetching implementation. 
I will have to change the code and i will add error handling when i do that.

File: components/shared/Mentor/mentor-container.tsx (hunk 4 above)
Comment (nathnael-desta) [reply components/shared/Mentor/mentor-container.tsx:25]: thanks, i have noted this issue. i will fix this when i completely redo the fetching implementation according to the new schema that the backend will send me/

**Review by misudesu — APPROVED** [review]

**Comment by misudesu** [comment]
ready to merge

**Pull request merged**